docs/source/deployment-experience.md

```diff
@@ -0,0 +1,135 @@
+# Deployment Experience
+
+Several facilities deploy Bluesky across many instruments (up to 25 or so).
+In this document we summarize lessons learned from that experience. At the
+moment, only NSLS-II (Bluesky's facility of origin) is represented, but we
+hope to add comments from other facilities in the future.
+
+## NSLS-II
+
+### Python environment management
+
+We deploy conda environments, verioned by operating cycle, in a central (NFS)
+software mount. The instrument staff and users have only read-only access to
+these environments. The environments contains the union of all requirements
+across all instruments: they include core scientific Python libraries (e.g.
+scipy), Bluesky libraries, and other more instrument-specific libraries, both
+internally and externally developed.
+
+The environments read-only in order to avoid accidentally breaking changes.
+To support short-term patching and minor variations betweeen instruments,
+we add an instrument-specific directory to the `PYTHONPATH` at each
+instrument respectively, granting instrument staff read/write permission
+on that directory. This enables them to install extra packages and upgrade
+or downgrade existing ones as needed, without the risk of unintended
+consequences for other instruments.
+
+The conda environments are build via a
+[CI service](https://github.com/nsls2-conda-envs),
+hosted on [Zenodo](https://zenodo.org/record/6555136), and
+deployed via conda-pack. Using conda-pack speeds up installation and
+improves reliability.
+
+We recommend:
+
+* Create centrally-managed read only conda enviornments.
+* Use `PYTHONPATH` to "overlay" custom additions or version changes.
+
+### Database Management
+
+NSLS-II and some other facilities we know of currently use a separate
+single-node MongoDB instance per instrument, but _we do not recommend this_. We
+are moving toward a central MongoDB cluster shared by all instrumenents. This
+reduces maintainence burden (e.g. less work to update MongoDB version) and
+improves reliability (avoid single-node deployments which are single points
+of failure).
+
+Each instrument will have its own MongoDB database on the cluster.
+
+We recommend:
+
+* Use a central cluster, shared by all instruments.
+* Use authentication between the MongoDB client and server.
+
```

Comment:
APS (currently) has switched to a set of MongoDB VMs, where several instruments will share a single VM.  The VMs are managed by the APS IT group.  The set of VMs is not clustered, since this would trigger an enormous cost increase.  We'll see how that goes.  Recommendations on how to cluster without large cost increase are welcome.